### ISSUE TITLE
Multichat with DiscordSRV and CMI

### ISSUE THREAD

**Issue by FuckingDeveloper** [Zrips/CMI#5631]
Hello. I noticed one very strange thing that I couldn't figure out. My friend, **Vainery**, wrote about it in **Discord CMI**.

The essence of the problem is as follows. I need to create a multichat from Discord and Minecraft. To do this, I use the build from **DiscordSRV** and the purchased **CMI plugin**.

I tried different configurations of the CHAT section in the CMI config. But I still don't understand how it works. Or rather, I was thinking... And whether it works correctly in CMI.

I found a feature that when trying to use a global chat (shouting system) and a local chat (traveler chat). My bundle works, but **DiscordSRV** does not want to capture at least one of the two chats (_cmi.chat.shout_ permission is active). But as soon as I disable _cmi.chat.shout_ in **luckperms**... Global chat via exclamation chat stops working in the game, but just **DiscordSRV** starts capturing these messages, sending them to Discord along with the exclamation mark. It does not process messages without an exclamation mark.
The reverse chat from Discord to CMI works perfectly. There are no complaints.

In addition to my question, I attach the plugin versions, the **CMI** config, **luckperms**, as well as the output of messages when _cmi.chat.shout_ is turned off.

I think this is a bug, but I'm not sure about it. Is there any solution to this problem?

![00FbG5MT2EE](https://user-images.githubusercontent.com/26286168/120069427-a620f980-c08e-11eb-8165-dccee0b775f4.jpg)
![vZALxj7PTGk](https://user-images.githubusercontent.com/26286168/120069431-a91bea00-c08e-11eb-8b43-721c6517bb26.jpg)
![P2kKV2YM9Gc](https://user-images.githubusercontent.com/26286168/120069441-ae793480-c08e-11eb-850f-94f74e7ed0f5.jpg)
![w0dz-U2Pbac](https://user-images.githubusercontent.com/26286168/120069444-b042f800-c08e-11eb-8211-264e9af39471.jpg)
![eEXZ9pmWm2c](https://user-images.githubusercontent.com/26286168/120069461-c0f36e00-c08e-11eb-99f8-4f7cf62c2d55.jpg)

**Comment by NeotenySS** [comment]
По Русский:

Если использовать DiscordSRV, без плагина CMI, то передача сообщений происходит корректно? 
Я так понимаю, что благодаря CMI используется правило в виде "!" которое способствует передачи сообщений в дискорд только если пременить его. Некоторым эта функция полезна чтобы дать выбор игроку отправить сообщение в чат сервера или отправить как в чат сервера, так и в чат Discord. Возможно такова задумка, но я прошу разработчика обратить на это внимание и если возможно, добавить функцию где игрок может не использовать "знак" для отправки сообщений одновременно в игровой чат и чат Discord. 
/---/

For English:

If you use DiscordSRV, without the CMI plugin, is the message passing correct?
I understand that thanks to CMI, a rule of the form "!" which promotes the transmission of messages to discord only if you change it. For some, this feature is useful to give the player the choice to send a message to the server chat or send to both the server chat and the Discord chat. Perhaps this is the idea, but I ask the developer to pay attention to this and, if possible, add a function where the player can not use the "sign" to send messages simultaneously to the game chat and the Discord chat.

**Comment by CloudeLecaw** [comment]
We also integrated our main chat into discord with using cmi chat it works fine that ! is not extra needed for that

**Comment by Mrhealman** [comment]
> По Русский:
> 
> Если использовать DiscordSRV, без плагина CMI, то передача сообщений происходит корректно?
> Я так понимаю, что благодаря CMI используется правило в виде "!" которое способствует передачи сообщений в дискорд только если пременить его. Некоторым эта функция полезна чтобы дать выбор игроку отправить сообщение в чат сервера или отправить как в чат сервера, так и в чат Discord. Возможно такова задумка, но я прошу разработчика обратить на это внимание и если возможно, добавить функцию где игрок может не использовать "знак" для отправки сообщений одновременно в игровой чат и чат Discord.
> /---/
> 
> For English:
> 
> If you use DiscordSRV, without the CMI plugin, is the message passing correct?
> I understand that thanks to CMI, a rule of the form "!" which promotes the transmission of messages to discord only if you change it. For some, this feature is useful to give the player the choice to send a message to the server chat or send to both the server chat and the Discord chat. Perhaps this is the idea, but I ask the developer to pay attention to this and, if possible, add a function where the player can not use the "sign" to send messages simultaneously to the game chat and the Discord chat.

Для русских:
Да, все работает прекрасно. DiscordSRV захватывает сообщения из игры и посылает их в чат без плагина CMI.
Если я ставлю плагин CMI, включаю режим shout через знак "!" (в привилегиях cmi.chat.shout запрещен), то DiscordSRV начинает захвать сообщения, но только те, которые я пишу через "!" (соответственно в игре глобальный чат не работает, а DiscordSRV думает, что все прекрасно и захватывает сообщения из глобального чата, НО захватывает их со знаком "!"). То есть интеграция с DiscordSRV работает, но работает наоборот, никак надо. Если же я даю игрокам право использовать cmi.chat.shout, то DiscordSRV перестает работать. 

For English:
Yes, everything works fine. DiscordSRV captures messages from the game and sends them to the chat without the CMI plugin.
If I install the CMI plugin, I enable the shout mode through the"! " sign (in the cmi.chat.shout privilege is disabled), then DiscordSRV starts capturing messages, but only the ones I write via "!" (accordingly, the global chat does not work in the game, and DiscordSRV thinks that everything is fine and captures messages from the global chat, BUT captures them with a"! " sign). That is, the integration with DiscordSRV works, but it works the other way around. If I give players the right to use cmi.chat.shout, DiscordSRV stops working.

**Comment by Mrhealman** [comment]
> We also integrated our main chat into discord with using cmi chat it works fine that ! is not extra needed for that
> you just need to figure the disorc part corecctly in the cmi settings

I can see that the integration works, but why it works the other way around. That is, the cmi.chat.shout privilege is prohibited, and DiscordSRV catches messages + puts a "!" sign at the beginning of them. Moreover, it does not take a local chat. But if I give players the option to use the global chat via shout, I give them the privilege to do so DiscordSRV stops working for both the local chat and the global chat.

**Comment by CloudeLecaw** [comment]
youdefine what is infront of a shout

**Comment by Mrhealman** [comment]
> youdefine what is infront of a shout
> also you define the discord format for normal messages
> 
> ```
>   DiscordSRV:
>     # Enables support for DiscordSRV plugin
>     Enabled: false
>     # Defines name of global chat channel in discordsrv
>     GlobalChannel: global
>     # Indicator which can be used as {discord} in chat format to indicate that message came from discord and not ingame
>     Label: '&2[&7DC&2]'
>     UnlinkedLabel: '&4[&cD&4]'
> ```
> 
> ```
>   ShoutRange: 0
>   # Defines cost for each shout message
>   ShoutCost: 0
>   # Prefix used to indicate that message should be sent to public chat instead of current players chat room
>   # Set it to empty field if you want this feature to be disabled
>   ChatRoomShout: '!'
> ```
> 
> and in langfile
> 
> ```
> Chat:
>   localPrefix: ''
>   shoutPrefix: '&c[S]&r'
> ```

You don't understand the problem, or you just don't want to understand it. I wrote that shout does not send messages to DiscordSRV if people have the cmi.chat.shout privilege enabled. If I disable this privilege, DiscrodSRV captures messages from the Minecraft chat and sends them to Discord, while adding "! " to the beginning of the message, but players cannot use shout with this configuration.

**Comment by CloudeLecaw** [comment]
I see so your Problem is shouts are not sendet to discord right?

**Comment by Mrhealman** [comment]
Yes, from the point of view of Discrod to Minecraft everything works, but from the point of view of Minecraft to Discord chat does not work. That's my problem.

**Comment by NeotenySS** [comment]
I think this bug 🐞
